storageExample.go

```diff
@@ -3,223 +3,229 @@ package main
 
 import (
 	"encoding/base64"
+	"flag"
 	"fmt"
 	"io/ioutil"
 	"math/rand"
 	"os"
 
+	"bytes"
+
 	"github.com/Azure/azure-sdk-for-go/storage"
 )
 
-type storageEnvironment string
+var (
+	accountName string
+	accountKey  string
+	blobCli     storage.BlobStorageClient
+	emulator    *bool
 
-const (
-	emulator storageEnvironment = "emulator"
-	account  storageEnvironment = "account"
+	appendBlobFile = "appendBlob.txt"
+	blockBlobFile  = "blockBlob.txt"
+	pageBlobFile   = "pageBlob.txt"
 )
 
+func init() {
+	emulator = flag.Bool("emulator", false, "use the Azure Storage Emulator")
+	flag.Parse()
+	if *emulator {
+		accountName = storage.StorageEmulatorAccountName
+		accountKey = storage.StorageEmulatorAccountKey
+	} else {
+		accountName = getEnvVarOrExit("ACCOUNT_NAME")
+		accountKey = getEnvVarOrExit("ACCOUNT_KEY")
+	}
+	client, err := storage.NewBasicClient(accountName, accountKey)
+	onErrorFail(err, "Create client failed")
+
+	blobCli = client.GetBlobService()
+
+	// delete all containers in the emulator
+	if *emulator {
+		list, err := blobCli.ListContainers(storage.ListContainersParameters{})
+		onErrorFail(err, "List containers failed")
+
+		for _, c := range list.Containers {
+			cnt := blobCli.GetContainerReference(c.Name)
+			err = cnt.Delete(nil)
+			onErrorFail(err, "Delete container failed")
+		}
+	}
+}
+
 func main() {
 	fmt.Println("Azure Storage Blob Sample")
-	err := blobSamples("demoblobconatiner", "demoPageBlob", "demoAppendBlob", "demoBlockBlob")
-	printError(err)
+	blobSamples("demoblobconatiner", "demoPageBlob", "demoAppendBlob", "demoBlockBlob")
 }
 
 // blobSamples creates a container, and performs operations with page blobs, append blobs and block blobs.
-func blobSamples(containerName, pageBlobName, appendBlobName, blockBlobName string) error {
-	fmt.Println("Get credentials...")
-	credentials, err := getCredentials(emulator)
-	// credentials, err := getCredentials(account)
+func blobSamples(containerName, pageBlobName, appendBlobName, blockBlobName string) {
+	fmt.Println("Create container with private access type...")
+	cnt := blobCli.GetContainerReference(containerName)
+	options := storage.CreateContainerOptions{
+		Access: storage.ContainerAccessTypePrivate,
+	}
+	err := cnt.Create(&options)
 	if err != nil {
-		return err
+		if accountName == storage.StorageEmulatorAccountName {
+			onErrorFail(err, "Create container failed: If you are running with the emulator credentials, plaase make sure you have started the storage emmulator. Press the Windows key and type Azure Storage to select and run it from the list of applications - then restart the sample")
+		}
+		onErrorFail(err, "Create container failed")
 	}
 
-	fmt.Println("Create storage client...")
-	client, err := storage.NewBasicClient(credentials["AZURE_STORAGE_ACCOUNT_NAME"], credentials["AZURE_STORAGE_ACCOUNT_KEY"])
-	if err != nil {
-		return err
+	if !*emulator {
+		err = appendBlobOperations(cnt, appendBlobName)
```

Comment:
Why does this happen only when not running under the emulator?  Probably good to add a comment.